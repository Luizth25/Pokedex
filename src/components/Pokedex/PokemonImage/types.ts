export type TPokemonImageProps = {
  image: string;
  name: string;
};
