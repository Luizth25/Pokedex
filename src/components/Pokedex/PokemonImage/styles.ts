import styled from "styled-components";

export const PokemonImageContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 10px;
  width: 150px;
  height: 150px;
  border-radius: 50%;
  overflow: hidden;
  box-shadow: inset -10px 10px 0 10px #ccc;
  background-color: #fff;
  border: 10px solid #000;
  position: relative;
`;
