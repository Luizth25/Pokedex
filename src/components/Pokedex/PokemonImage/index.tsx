import { TPokemonImageProps } from "./types";
import * as s from "./styles";

const PokemonImage = ({ image, name }: TPokemonImageProps) => (
  <s.PokemonImageContainer>
    <img src={image} alt={name} />
  </s.PokemonImageContainer>
);

export default PokemonImage;
