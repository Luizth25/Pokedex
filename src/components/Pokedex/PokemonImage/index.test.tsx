import { render, screen } from "@testing-library/react";

import PokemonImage from ".";

describe("The PokemonImage Component", () => {
  it("Should be defined", () => {
    expect(PokemonImage).toBeDefined();
  });

  it("Should have Image", () => {
    render(<PokemonImage image="img/img.png" name="image" />);

    expect(screen.getByRole("img")).toHaveAttribute("src", "img/img.png");
  });
});
