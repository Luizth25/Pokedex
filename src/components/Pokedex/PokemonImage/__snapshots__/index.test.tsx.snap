// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PokemonImage Component Should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe dFsVnp"
  >
    <img
      alt="image"
      src="img/img.png"
    />
  </div>
</div>
`;
